communication/src/spark_protocol_functions.cpp

```diff
@@ -196,41 +196,72 @@ int spark_protocol_set_connection_property(ProtocolFacade* protocol, unsigned pr
         void* reserved) {
     ASSERT_ON_SYSTEM_THREAD();
     switch (property) {
-    case particle::protocol::Connection::PING: {
-        const auto d = (const particle::protocol::connection_properties_t*)data;
+    case Connection::PING: {
+        const auto d = (const connection_properties_t*)data;
         protocol->set_keepalive(value, d->keepalive_source);
         return 0;
     }
-    case particle::protocol::Connection::FAST_OTA: {
+    case Connection::FAST_OTA: {
         protocol->set_fast_ota(value);
         return 0;
     }
-    case particle::protocol::Connection::DEVICE_INITIATED_DESCRIBE: {
+    case Connection::DEVICE_INITIATED_DESCRIBE: {
         protocol->enable_device_initiated_describe();
         return 0;
     }
-    case particle::protocol::Connection::COMPRESSED_OTA: {
+    case Connection::COMPRESSED_OTA: {
         protocol->enable_compressed_ota();
         return 0;
     }
-    case particle::protocol::Connection::SYSTEM_MODULE_VERSION: {
+    case Connection::SYSTEM_MODULE_VERSION: {
         protocol->set_system_version(value);
         return 0;
     }
-    case particle::protocol::Connection::MAX_BINARY_SIZE: {
+    case Connection::MAX_BINARY_SIZE: {
         protocol->set_max_binary_size(value);
         return 0;
     }
-    case particle::protocol::Connection::OTA_CHUNK_SIZE: {
+    case Connection::OTA_CHUNK_SIZE: {
         protocol->set_ota_chunk_size(value);
         return 0;
     }
-    case particle::protocol::Connection::MAX_TRANSMIT_MESSAGE_SIZE: {
+    case Connection::MAX_TRANSMIT_MESSAGE_SIZE: {
         protocol->set_max_transmit_message_size(value);
         return 0;
     }
     default:
-        return particle::protocol::ProtocolError::NOT_IMPLEMENTED;
+        return ProtocolError::NOT_IMPLEMENTED;
+    }
+}
+
+int spark_protocol_get_connection_property(ProtocolFacade* protocol, unsigned property, void* data, size_t* size, void* reserved)
+{
+    ASSERT_ON_SYSTEM_THREAD();
```

Comment:
This shouldn't work for threaded applications as of right with how this is used in `CloudClass`. Add `spark_get_connection_property()` to schedule to run on system thread?